proxy/src/connection.rs

```diff
@@ -110,6 +142,24 @@ pub struct PeekFuture<T> {
     inner: Option<T>,
 }
 
+impl TlsConnect {
```

Comment:
1. These two types are the same type so they should be represented by one type parameterized by `C` which will be either `ClientConfig` or `ServerConfigWatch`.

2. Would it make sense to just add `sensor` to `tls::ConnectionConfig` and avoid these two new types? Or use `tls::ConnectionConfig<(C, Sensor)>`? I noticed we're already using a tuple `(Sensor, C)` in other parts of the code below.

Reply:
> 1. These two types are the same type so they should be represented by one type parameterized by `C` which will be either `ClientConfig` or `ServerConfigWatch`.

They'd have to be parameterized over the sensor type as well. At the time, I thought it seemed less complex to just have two separate types, but that can be changed.

> 2. Would it make sense to just add `sensor` to `tls::ConnectionConfig` and avoid these two new types?

That was actually what I initially considered doing. I thought that (due to the name) the intention behind the `ConnectionConfig` type was to hold only config, and not things with more complex behaviours like the sensor, and thus figured I'd probably be asked to move the sensor _out_ of `ConnectionConfig`. But, if you think that would be better, we could do that.